client/app/utils/users/adminUserOperators.ts

```diff
@@ -49,10 +50,9 @@ export async function processExternalDashboardUsersTileData() {
   const selfIndex = transformedTileData.findIndex((userOperator) => {
     return userOperator.user.user_guid.replace(/-/g, "") === uid;
   });
-
   // Ensure selfIndex is within the valid range before modifying the array
   if (selfIndex !== -1 && selfIndex < transformedTileData.length) {
-    transformedTileData[selfIndex].status = Status.MYSELF;
+    transformedTileData[selfIndex].isUserCurrentUser = true;
```

Comment:
In client/app/components/datagrid/cells/ChangeUserOperatorStatusColumnCell.tsx, we have a `buttonsToShow` function that shows the approve/reject/undo buttons based on the statuses. This means that as soon as I changed "Myself" to "Approved", the buttons showed up. 

This file is the only place we have access to the user_guid (this function strips out the user_guid and anything else that shouldn't show in the table). We needed some way to keep track of who the current user is besides the "Myself" status, so I've added an `isUserCurrentUser` key. This key is passed to the grid but is hidden in the table. I'm not super happy with this solution, but wasn't able to come up with anything better. Thoughts very welcome!

Reply:
Wouldn't it be far simpler to just change the name in the status pill itself? In `datagrid/helpers.tsx`:

```
  const status = (params.value = Status.MYSELF
    ? Status.APPROVED
    : (params.value as string));
```

As an aside, I think the `statusStyle` cell in `helpers` should be moved to its own file in `/cells` to be easier to find 😅